src/global/reco/reco.cc

```diff
@@ -77,12 +77,27 @@ void InitPlugin(JApplication* app) {
        "EcalEndcapPClusterAssociations"},
       {"EcalClusterAssociations"}, app));
 
+  // Combine reconstructed charged particles and their associations from different sources
+  app->Add(
+      new JOmniFactoryGeneratorT<CollectionCollector_factory<edm4eic::ReconstructedParticle, true>>(
+          "CombinedReconstructedChargedParticles",
+          {"ReconstructedChargedParticles", "TaggerTrackerReconstructedParticles"},
```

Comment:
This changes definition of `ReconstructedChargedParticles` to not include FB particles. We probably don't want that.

Reply:
There are no clean ways of doing this. Assuming we don't want to replace `ReconstructedChargedParticles` with a subset collection we need to pass it though another algorithm first, this would mean merging it before the final `PID` factory, which may be the only reasonable approach but feels unnecessary.

Reply:
If I understand correctly, the issue is with needing association rewriting. Letting pid_lut algorithm do the rewriting for you is one way. In one of the PRs on calorimetry we've envisioned a generic association rewriting algorithm for such use cases, but wasn't implemented so far.